apps/src/code-studio/components/ShareDialog.jsx

```diff
@@ -160,94 +190,136 @@ var ShareDialog = React.createClass({
+                styles={{label:{marginTop: 15, marginBottom: 0}}}
+              />}
+              {(this.props.appType === 'applab' || this.props.appType === 'gamelab') &&
+              <AdvancedShareOptions
+                i18n={this.props.i18n}
+                shareUrl={this.props.shareUrl}
+                onClickExport={this.props.onClickExport}
+                expanded={this.state.showAdvancedOptions}
+                onExpand={this.showAdvancedOptions}
+                channelId={this.props.channelId}
+                embedOptions={embedOptions}
+              />}
+              {/* Awkward that this is called continue-button, when text is
+               close, but id is (unfortunately) used for styling */}
+              <button
+                id="continue-button"
+                style={{position: 'absolute', right: 0, bottom: 0, margin: 0}}
+                onClick={this.close}
+              >
+                {this.props.i18n.t('project.close')}
+              </button>
             </div>
-            {this.state.showSendToPhone &&
-             <SendToPhone
-               channelId={this.props.channelId}
-               appType={this.props.appType}
-               styles={{label:{marginTop: 15, marginBottom: 0}}}
-             />}
-            {(this.props.appType === 'applab' || this.props.appType === 'gamelab') &&
-             <AdvancedShareOptions
-               i18n={this.props.i18n}
-               shareUrl={this.props.shareUrl}
-               onClickExport={this.props.onClickExport}
-               expanded={this.state.showAdvancedOptions}
-               onExpand={this.showAdvancedOptions}
-               channelId={this.props.channelId}
-               embedOptions={embedOptions}
-             />}
-            {/* Awkward that this is called continue-button, when text is
-            close, but id is (unfortunately) used for styling */}
-            <button
-              id="continue-button"
-              style={{position: 'absolute', right: 0, bottom: 0, margin: 0}}
-              onClick={this.close}
-            >
-              {this.props.i18n.t('project.close')}
-            </button>
           </div>
-        </div>
-      </BaseDialog>
+        </BaseDialog>
+        <PublishDialog/>
+      </div>
+
     );
   }
 });
-module.exports = ShareDialog;
+
+export default connect(state => ({
+  isOpen: state.shareDialog.isOpen,
+}), (dispatch, ownProps) => ({
+  onClose: () => dispatch(hideShareDialog()),
+  onShowPublishDialog() {
+    // Allow onShowPublishDialog to be overridden when passed explicitly as a prop
+    if (ownProps.onShowPublishDialog) {
+      ownProps.onShowPublishDialog();
+    } else {
+      dispatch(hideShareDialog());
+      dispatch(showPublishDialog(ownProps.channelId, ownProps.appType));
+    }
+  },
+  onUnpublish() {
+    if (ownProps.onUnpublish) {
+      ownProps.onUnpublish();
+    } else {
+      dispatch(unpublishProject(ownProps.channelId));
+    }
+  },
```

Comment:
Simply moving this logic into the component isn't an option because it won't have access to ownProps there. [mergeProps](https://github.com/reactjs/react-redux/blob/master/docs/api.md#arguments) to the rescue!